RFC Draft.md

```diff
@@ -535,6 +535,111 @@ function subscribeToCreate() {
 }
 ```
 
+## Flatten the GraphQL operations' responses
+As alluded to in the prior suggestion, we're looking to flatten the results for the GraphQL operations to make them more easily accessible instead of the current three-levels-deep nested object. Would love to get your understanding on which option you prefer.
+
+**Current Usage (v5)**
+```ts
+async function createNewTodo() {
+  const res: GraphQLResult<Todo> = await API.graphql<GraphQLQuery<Todo>>(
+    graphqlOperation(createTodo, {
+      input: { id: uuid() },
+    })
+  );
+  // todo result is nested
+  console.log(res.data.createTodo); 
+}
+
+interface GraphQLResult<T = object> {
+    data?: T; // in the above example T is Todo
+    errors?: GraphQLError[];
+    extensions?: {
+        [key: string]: any;
+    };
+}
+```
+
+### Proposed behavior for single query/mutation in the response
+**Proposed Option 1: Flatten to the lowest level (v6)**
+```ts
+async function createNewTodo() {
+  const res: Todo = await API.graphql<GraphQLQuery<Todo>>(
+    graphqlOperation(createTodo, {
+      input: { id: uuid() },
+    })
+  );
```

Comment:
```suggestion
  const res: Todo = await API.mutate(createTodo, {
    input: { id: uuid() },
  });
```

These v6 examples should probably use the proposed DX from the previous section, or do you think they should be kept separate?